src/authentication.jl

```diff
@@ -515,6 +515,8 @@ function _juliahub_project(
     elseif isa(project, UUIDs.UUID)
         return project
     elseif isa(project, AbstractString)
+        isempty(project) && return nothing
```

Comment:
I think it would be better to add this only for the actual environment variable above. I.e. if the user passes an explicit empty string, we should still error. But let's `isempty(strip(get(ENV, "JULIAHUB_PROJECT_UUID", ""))) -> nothing` the environment variable?